From 6b5222bd5ad6a3a20803640b6fc25917b76581b4 Mon Sep 17 00:00:00 2001
From: Yangbo Lu <yangbo.lu@nxp.com>
Date: Mon, 25 Apr 2016 17:53:07 +0800
Subject: [PATCH 389/666] mmc: sdhci-of-esdhc: fix DAT/CMD signal level
 checking

This patch fixes DAT/CMD signal level checking issue.

Signed-off-by: Yangbo Lu <yangbo.lu@nxp.com>
[Original patch taken from QorIQ-SDK-V2.0-20160527-yocto]
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 drivers/mmc/host/sdhci-of-esdhc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mmc/host/sdhci-of-esdhc.c b/drivers/mmc/host/sdhci-of-esdhc.c
index a353455..3590aa3 100644
--- a/drivers/mmc/host/sdhci-of-esdhc.c
+++ b/drivers/mmc/host/sdhci-of-esdhc.c
@@ -137,7 +137,7 @@ static u32 esdhc_readl_fixup(struct sdhci_host *host,
 		dlsl = value & ESDHC_DLSL_MASK;
 		ret = value &
 		      (~((ESDHC_CLSL_MASK << 1) | (ESDHC_DLSL_MASK >> 4)));
-		ret = value | ((clsl << 1) | (dlsl >> 4));
+		ret |= ((clsl << 1) | (dlsl >> 4));
 		return ret;
 	}
 	ret = value;
-- 
2.7.4

